File: 1.3-cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int total_coins(int value);

int main(void)
{
    // Repete se não for um valor positivo.
    double dollar, cents;
    float value;
    int coins;

    do
    {
        value = get_float("Value: ");
    }
    while (value <= 0.00);

    // Checa se tem apenas centavos ou não.
    cents = modf(value, &dollar);
    cents = round(cents * 100);

    // Trata (ou não) os centavos se não tiver dolares.
    if (dollar <= 0)
    {
        int total_cents = (int) cents;
        coins = total_coins(total_cents);
        printf("%i\n", coins);
    }
    else
    {
        int total_cents = (int) dollar * 100;
        total_cents = total_cents + cents;
        coins = total_coins(total_cents);
        printf("%i\n", coins);
    }
}

int total_coins(int value)
{
    int coins = 0;
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int pennie = 1;

    // Compara os valores para saber
    // quantas moedas serão necessárias.
    while (value > 0)
    {
        if (value >= quarter)
        {
            coins += 1;
            value -= quarter;
        }
        else if (value >= dime)
        {
            coins += 1;
            value -= dime;
        }
        else if (value >= nickel)
        {
            coins += 1;
            value -= nickel;
        }
        else
        {
            coins += 1;
            value -= pennie;
        }
    }

    return coins;
}

File: 6.3 cash/cash.py
from cs50 import get_float


def main():
    cash = set_cash()
    coins = 0

    while cash > 0:

        if cash >= 0.25:
            cash -= 0.25
            cash = round(cash, 2)
            coins += 1

        elif cash >= 0.10:
            cash -= 0.10
            cash = round(cash, 2)
            coins += 1

        elif cash >= 0.05:
            cash -= 0.05
            cash = round(cash, 2)
            coins += 1

        else:
            cash -= 0.01
            cash = round(cash, 2)
            coins += 1

    print(coins)

def set_cash():
    cash = get_float("Change owed: ")

    if cash < 0:
        cash = set_cash()

    return cash

main()